### ISSUE TITLE
About Visual Head Chase

### ISSUE THREAD

**Issue by cfmaXT** [CR400AF-A/SparseMM#11]
Thanks for the excellent work!

I have a question about the visual head chase. The question concerns the function find_image_idx(image_size, image_feature, parsed_data) in llava_visual_head.py. 

For the code:
        flag = torch.zeros(1, image_feature.shape[1], image_feature.shape[2]).to(image_feature.device)
        bbox = parse_data[1]
        bbox_width = (bbox[0][0], bbox[1][0])
        bbox_height = (bbox[0][1], bbox[1][1])
        start_width, end_width = bbox_width[0] // mapping_pixel_width, bbox_width[1] // mapping_pixel_width + 1
        start_height, end_height = bbox_height[0] // mapping_pixel_height, bbox_height[1] // mapping_pixel_height + 1
        flag[:, int(start_height):int(end_height), int(start_width):int(end_width)] = 1
        flag = torch.cat((flag, torch.zeros(1, flag.shape[1], 1).to(image_feature.device)), dim=-1) # image new line
        # import pdb; pdb.set_trace()
        flat_flag = flag.flatten(1,2)
        flag_idx = torch.where(flat_flag == 1)[1]
        flag_idxs.append((parse_data,flag_idx))

Since the flag_idxs will be utilized to determine hit_score, I think the order of elements in flag_idxs should be consistent with the image token order of LLM. I want to know whether the function (e.g., flat_flag = flag.flatten(1,2)) considers the order of multiple slices in high-resolution images?

Thanks for your help!

**Comment by CR400AF-A** [comment]
Sorry for the late reply. 

The order of the flag_idxs is correct. Since the image_feature has already been correctly concatenated into a large feature map in the 2D dimension, it can be directly flattened. You can refer [this](https://github.com/CR400AF-A/SparseMM/blob/main/visual_head/LLaVA-NeXT/llava/model/llava_arch.py#L298).

**Comment by CR400AF-A** [comment]
Close this issue.
Please feel free to reopen it if the problem persists.
